New settings screen to activate each connected app type

dacook opened the pull request:
**Funded feature code is _#12543 [DFC] Anonymized orders endpoint_**

#### What? Why?

- Closes #12549

I forgot to make the  new admin settings screen hidden behind the `connected_apps` feature toggle. But after this issue we will be able to remove the toggle anyway, so it's probably ok.

#### What should we test?
As per issue. In addition:
* ensure `connected_apps` feature toggle is enabled

Note: Both connected app types are disabled by default, so the Enterprise settings tab will not show.
_Except_ if there are enterprises with connected apps on the server already: then they will be enabled in Connected Apps Settings (to avoid them becoming unexpectedly hidden).


#### Release notes

<!-- Please select one for your PR and delete the other. -->

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [ ] Feature toggled

<!-- Choose a pull request title above which explains your change to a
     a user of the Open Food Network app. -->

The title of the pull request will be included in the release notes.


#### Dependencies
<!-- Does this PR depend on another one?
     Add the link or remove this section. -->



#### Documentation updates
<!-- Are there any wiki pages that need updating after merging this PR?
     List them here or remove this section. -->

mkllnk commented on db/migrate/20240801034710_set_connected_apps_enabled_if_any.rb:
```diff
@@ -0,0 +1,14 @@
+  end
+end
+
+class ConnectedApp < ApplicationRecord
+  scope :discover_regen, -> { where(type: "ConnectedApp") }
+  scope :affiliate_sales_data, -> { where(type: "ConnectedApps::AffiliateSalesData") }
+end
```
We usually define these as sub-classes of the migration. The table name still works and it's definitely another namespace to the original model.

Thankfully, good_migrations ensures that we don't load the actual model here.

mkllnk reviewed (approved):
Great simple implementation. :+1:

rioug reviewed (approved):
Looks good :+1:

RachL commented:
> it won't have any effect, so I don't think it's worth advertising.

I agree :+1:

however @dacook I've tried 3 times on both FR and AU and I can't stage the pr :/

rioug commented:
There were some good_migration error in bugsnag, ie: https://app.bugsnag.com/yaycode/openfoodnetwork-aus/errors/66b0ce0c5e9c074bc4c15904?event_id=66b0ce0c00f6a8c721040000&i=sk&m=nw 
Looks like we need to fix the migration, per Maikel's comment

dacook commented:
@RachL ready to test on au_staging! If you are unable to test straight away, please remove stagin label so that Konrad is able to use it.

RachL commented:
I'm on it, I will try to do it in one go :+1:

RachL commented:
> ensure connected_apps feature toggle is enabled

I disabled it before the PR was stage, and it did not re-enabled it. But I think this is normal. In any case we will be able to update it manually in production.

> Enterprise settings will not show until they are enabled

This is ok ✅ 

Thank you so much for handling this case @dacook I realize I've forgotten about it when writing the issue:

![image](https://github.com/user-attachments/assets/0615e732-72a8-4fed-9d30-29b6dda41d3c)

I guess super admins should be able to do it, but it's a minor problem and can be dealt with in a different request (which I will open an issue for only if this becomes a problem).

However:

> Disabling disco regen does not seem possible

I ended up with a white page and the app remains available in enterprise settings. @dacook Is it because it's enable by default on AU with the old config? In that case it won't be a problem in production and we can merge.

![image](https://github.com/user-attachments/assets/ea2ecf09-aac0-49e1-aeae-fc2f701db02b)

dacook commented:
Thanks @RachL , actually you uncovered a bug, oops! I have fixed this now, and added a success message when saving:
![Screenshot 2024-08-07 at 9 44 58 am](https://github.com/user-attachments/assets/1a5174d0-de6a-4f18-9db4-1cbc155b615d)

But I have discovered something else that I might need to do for this PR, see comment on issue.

dacook commented:
Two new commits for review.

RachL commented:
Awesome, LGTM! Many thanks @dacook :pray: 

Merging :muscle:
